apisix/plugins/hmac-auth.lua

```diff
@@ -244,7 +244,13 @@ local function generate_signature(ctx, secret_key, params)
 
             -- whether to encode the uri parameters
             if type(param) == "table" then
+                local vals = {}
                 for _, val in pairs(param) do
+                    core.table.insert(vals, val)
```

Comment:
Remember to convert boolean to `''` before sorting. `table.sort` only supports sorting in the same type.

Reply:
done